### PR TITLE
validate-modules - Use error codes instead of numbers

### LINKED ISSUE #59405: Use error codes instead of numbers in validate-modules
### SUMMARY

Update validate-modules to report messages using named error codes like `version-added` instead of error numbers like `E305`.

This would match the behavior of other tools we use, such as pylint, which have one named error code per rule. This will also make ignore entries much easier to read.

Doing this before 2.9 is released will avoid potential backwards compatibility issues with ignore entries in collections.

##### ISSUE TYPE

Feature Idea

##### COMPONENT NAME

validate-modules

### PR DESCRIPTION
**Pull request by samdoran**
##### SUMMARY

Add names for each of the error codes used by `validate-modules`.
Change the display message to include `name (code)`

Fixes #59405

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

`test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py`

### CODE REVIEW

**Comment by samdoran** [comment]
These names are highly up for debate. I attempted to stick with the mindset of describing the check not the error. Feel free to leave suggestions for improving the error code names.

**Comment by mattclay** [comment]
I don't see any reason to keep the existing numeric codes. We can just replace them with the names.

File: test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py [hunk 1]
```diff
@@ -501,13 +510,15 @@ def _find_module_utils(self, main):
                             if self._is_new_module():
                                 self.reporter.error(
                                     path=self.object_path,
+                                    name='FIXME',
```
Comment (mattclay) [inline test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py]: I assume this and the other `FIXME` names are placeholders for names not yet decided?

File: test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py [hunk 2]
```diff
@@ -654,6 +671,7 @@ def _ensure_imports_below_docs(self, doc_info, first_callable):
                         if future_import.name not in self.WHITELIST_FUTURE_IMPORTS:
                             self.reporter.error(
                                 path=self.object_path,
+                                name='allowed-future-imports',
```
Comment (mattclay) [inline test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py]: This test should be removed from the module validator. We have a future boilerplate sanity check which covers this. I'd remove it in a separate PR though.

File: test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py [hunk 3]
```diff
@@ -918,6 +948,7 @@ def _validate_docs(self):
                     # for error in errors:
                     #     self.reporter.error(
                     #         path=self.object_path,
+                    #         name='ansible-metadata-format',
```
Comment (mattclay) [inline test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py]: Was the YAML feature removed from the spec? If so, this whole commented block of code can be removed in a separate PR.

File: test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py [hunk 4]
```diff
@@ -362,6 +364,7 @@ def _check_interpreter(self, powershell=False):
         if not self.text.startswith('#!/usr/bin/python'):
             self.reporter.error(
                 path=self.object_path,
+                name='interpreter-line-python',
```
Comment (mattclay) [inline test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py]: I think we can remove this check in a separate PR, as the shebang sanity test covers this.

File: test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py [hunk 5]
```diff
@@ -354,6 +355,7 @@ def _check_interpreter(self, powershell=False):
             if not self.text.startswith('#!powershell\n'):
                 self.reporter.error(
                     path=self.object_path,
+                    name='interpreter-line-powershell',
```
Comment (mattclay) [inline test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py]: I think we can remove this check in a separate PR, as the shebang sanity test covers this.

File: test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py (hunk 1 above)
Comment (samdoran) [reply test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py]: Yes.

**Comment by mattclay** [comment]
Sanity tests passed. Other failures are unrelated. Merging quickly to avoid more rebase issues.
